docs/v0.5.x/upgrading-plugins.md

```diff
@@ -2,55 +2,60 @@
-
-`this.log` takes a second parameter `options`, you can pass `{ verbose: true }` to ensure that some log messages are always displayed during pipeline execution
-
-### For index/store adapters:
-1. instruct your users to install ember-cli-deploy-revision-data (see above)
-2. mind you that the revision key is now under `context.revisionData.revisionKey` (provided ember-cli-deploy-revision-data is installed) and it doesn't include your project name yet.
-
-```javascript
-_key: function(context) {
-  var revisionKey = context.commandOptions.revision || context.revisionData.revisionKey.substr(0, 8);
-  return context.project.name() + ':' + revisionKey;
-}
-```
-
-3. mind you that you won't be passed the contents of index.html to your upload function, instead you will have to read the file using the filesystem package,
-
-```javascript
-var path      = require('path');
-var fs        = require('fs');
-var readFile  = denodeify(fs.readFile);
-
-upload: function() {
-  readFile(path.join(context.distDir, "index.html"))
-  .then(function(buffer) {
-    return buffer.toString();
-  }).then(function(indexContents) {
-    // do uploady stuff with contents here
-  });
-}
-```
-
-### For asset adapters:
-1. if you have a hard reference to "tmp/asset-sync", replace that with `context.distDir`.
+Ember CLI Deploy 0.5.x introduces the concept of a deploy pipeline. Instead of writing *adapters* (like you did in 0.4.x), you write *plugins*, and then compose these plugins to configure your application's deploy pipeline. 
+
+While adapters had specific types (index adapter, asset adapter, etc.), plugins are generic, and are defined only by the [pipeline hooks](./pipelin-hooks) they implement.
+
+The changes you need to make to upgrade your adapter depends on its type.
+
+### Changes for all adapters
+
+  1. Add "ember-cli-deploy-plugin" to your addon's `package.json`
+
+  2. Install the [base plugin](https://github.com/ember-cli-deploy/ember-cli-deploy-plugin), and update your adapters to subclass it
+
+  3. Implement `createDeployPlugin` in `index.js` and return a subclass of `ember-cli-deploy-plugin`
+
+  4. Use the pipeline hooks to do the actual work:
+      - move uploading logic (from your index/asset adapters) to the `upload` hook
+      - move activation logic (from your index adapter) to the `activate` hook
+
+  5. Instruct your users to [update their config](./configuration-overview) and install [ember-cli-deploy-build](https://github.com/ember-cli-deploy/ember-cli-deploy-build)
+
+  6. Replace `console.log` statements with `this.log` to play nicely with Ember CLI Deploy's messaging
+
+      - Messages passed to `this.log` are displayed only when the verbose flag (`--verbose`) is present. You can pass `{ verbose: true }` as the second param to `this.log` to force your message to always display during pipeline execution.
```

Comment:
something like `always: true`? I get `verbose` is saying, "this is a verbose message, always show it", but because of the `verbose` flag it feels like "this is a verbose message, only show it during --verbose runs". Too ambiguous imo.

always: true
verboseOnly: false (default is true)
important: true